QEfficient/base/modeling_qeff.py

```diff
@@ -5,96 +5,307 @@
+        example_inputs: Dict[str, torch.Tensor],
+        output_names: List[str],
+        dynamic_axes: Dict[str, Dict[int, str]],
+        export_kwargs: Dict[str, any] = {},
+        onnx_transform_kwargs: Dict[str, any] = {},
+        export_dir: Optional[str] = None,
+    ) -> str:
+        """
+        Export the Pytorch model to ONNX.
+
+        Args:
+            :example_inputs (dict): Sample inputs to trace the model.
+            :output_names (list): names to assign to the output nodes of the graph, in order.
+            :dynamic_axes (dict): Same as dynamic_axes parameter to be passed to `torch.onnx.export`.
+            :export_kwargs (dict): Additional arguments to be passed to `torch.onnx.export`.
+            :onnx_transform_kwargs (dict): Additional arguments to be passed to `Transform.apply` for this class.
+            :export_dir (str): Specify the export directory. The export_dir will be suffixed with a hash corresponding to current model.
+        """
+        export_dir = Path(export_dir or (QEFF_HOME / self.model_name))
+        export_dir = export_dir.with_name(export_dir.name + "-" + self.model_hash)
+        onnx_path = export_dir / f"{self.model_name}.onnx"
+        if onnx_path.is_file():
+            self.onnx_path = onnx_path
+            return onnx_path
+
+        tmp_onnx_dir = export_dir / "onnx_tmp"
+        tmp_onnx_path = tmp_onnx_dir / f"{self.model_name}.onnx"
+        tmp_onnx_dir.mkdir(parents=True, exist_ok=True)
+
+        # Create input_names from example_inputs
+        input_names = []
+        for param in inspect.signature(self.model.forward).parameters:
+            if param in example_inputs:
+                if param == "past_key_values":
+                    for i in range(len(example_inputs["past_key_values"])):
+                        input_names.append(f"past_key.{i}")
+                        input_names.append(f"past_value.{i}")
+                else:
+                    input_names.append(param)
+
+        try:
+            torch.onnx.export(
+                self.model,
+                (example_inputs,),
+                str(tmp_onnx_path),
+                input_names=input_names,
+                output_names=output_names,
+                dynamic_axes=dynamic_axes,
+                opset_version=13,
+                **export_kwargs,
+            )
+            logger.info("Pytorch export successful")
+
+            model = onnx.load(tmp_onnx_path, load_external_data=False)
+            onnx_transform_kwargs = {
+                "onnx_base_dir": str(tmp_onnx_dir),
+                "model_name": self.model_name,
+                **onnx_transform_kwargs,
+            }
+            for transform in self._onnx_transforms:
+                model, transformed = transform.apply(model, **onnx_transform_kwargs)
+            model.metadata_props.append(
+                onnx.StringStringEntryProto(key="qeff_transforms", value=",".join(self._transform_names()))
+            )
```

Comment:
I'll add a TODO, but we can't really track that whether transform needs to be applied or not. It will be determined by the Transform. If we add the logic of determining if it needs to be applied, we go back to same `if-else` kind of applying transforms.